website/docs/tutorials/auth-token-example.md

```diff
@@ -2,3 +2,235 @@
 title: Passing Auth Token from Native to Portal
 sidebar_label: Passing Auth Token from Native to Portal
 ---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+A common scenario that a developer might run into is having a web experience tailored for the current logged in user.  Another scenario might be having a log-in screen be a Portal so it can easily be designed and updated across Android and iOS. In both of these scenarios, a developer would need to account for handling auth tokens between native and web code. Below are a few examples of how to solve these problems.
+
+## Passing from Native to Web
+
+When showing a Portal after a user has logged in, there are a few different ways to pass user auth tokens to a Portal.
+- Using the `PortalBuilder.setInitialContext()` function to set the initial state of the Portal.
+- Using the `PortalsPublish` to publish a message to the web app with the current auth tokens.
+- Using a custom plugin to send data back and forth from native and web.
+
+### Using PortalBuilder.setInitialContext()
+
+The easiest way to set the Portal's auth tokens is to use the `setInitialContext` function. Using this function will allow you to pass data that can be read almost immediately in the Portal.
+
+<Tabs 
+    defaultValue="swift" 
+    values={[
+        { label: 'Swift', value: 'swift', },
+        { label: 'Kotlin', value: 'kt', },
+        { label: 'Java', value: 'java', },
+    ]}
+>
+
+<TabItem value="swift">
+
+```swift {5}
+PortalManager.newPortal("user_page")
+    .setStartDir("web")
+    .setInitialContext([
+        "route": "/user",
+        "auth": /* Auth Data */
+    ])
+    .create()
+```
+
+</TabItem>
+
+<TabItem value="kt">
+
+```kotlin {5}
+PortalManager.newPortal("user_page")
+    .setStartDir("web")
+    .setInitialContext(mapOf(
+        "route" to "/user",
+        "auth" to /* Auth Data */,
+    ))
+    .create()
+```
+
+</TabItem>
+
+<TabItem value="java">
+
+```java {5}
+PortalManager.newPortal("user_page")
+    .setStartDir("web")
+    .setInitialContext(Map.ofEntries(
+        new AbstractMap.SimpleEntry<String, @NotNull Object>("route", "/user"),
+        new AbstractMap.SimpleEntry<String, @NotNull Object>("auth", /* Auth Data */)
+    ))
+    .create();
+```
+
+</TabItem>
+
+</Tabs>
+
+Then, in the entry point to your web application, you can use `Portals.getInitialContext()` to read the data passed in and act on it.
+
+```typescript title=main.ts
+import Portals from '@ionic/portals';
+
+type MyPortalContext = { route: string, auth: any };
+Portals.getInitialContext<MyPortalContext>().then(context => {
+    const auth = context.value.auth;
+    // rest of the web app...
+});
+```
+
+### Using the built in PortalsPlugin pub/sub functions
```

Comment:
I'm not sure about using pub/sub for this. Like you stated, if the web view isn't loaded and subscribed to the event, they won't get the message and when the web view loads it won't work. Makes me think we don't need this section.

Reply:
Yeah, I was iffy on adding it but figured I'd add it just in case. I'll go ahead and remove it and re-arrange the wording on the other section